numba/cuda/tests/nocuda/test_function_resolution.py

```diff
@@ -0,0 +1,37 @@
+from numba.cuda.testing import unittest, skip_on_cudasim
+import operator
+from numba.core import types, typing
+from numba.cuda.cudadrv import nvvm
+
+
+@unittest.skipIf(not nvvm.is_available(), "No libNVVM")
+@skip_on_cudasim("Skip on simulator due to use of cuda_target")
+class TestFunctionResolution(unittest.TestCase):
+    def test_fp16_binary_operators(self):
+        from numba.cuda.descriptor import cuda_target
+        ops = (operator.add, operator.iadd, operator.sub, operator.isub,
+               operator.mul, operator.imul)
+        for op in ops:
+            fp16 = types.float16
+            typingctx = cuda_target.typing_context
+            typingctx.refresh()
+            fnty = typingctx.resolve_value_type(op)
+            out = typingctx.resolve_function_type(fnty, (fp16, fp16), {})
+            if out != typing.signature(fp16, fp16, fp16):
+                raise AssertionError(out)
```

Comment:
Should this be `self.assertEqual(out, typing.signature(fp16, fp16, fp16), msg=str(out))` or similar? (same for the other test in this class).

Reply:
Yes.